ksqldb-engine/src/test/java/io/confluent/ksql/function/udf/string/RPadTest.java

```diff
@@ -18,69 +18,134 @@
-  public void shouldReturnNullForNullPadding() {
+  public void shouldReturnNullForNullInputBytes() {
+    final ByteBuffer result = udf.rpad(null, 4, BYTES_45);
+    assertThat(result, is(nullValue()));
+  }
+
+  @Test
+  public void shouldReturnNullForNullPaddingString() {
     final String result = udf.rpad("foo", 4, null);
     assertThat(result, is(nullValue()));
   }
 
   @Test
-  public void shouldReturnNullForEmptyPadding() {
+  public void shouldReturnNullForNullPaddingBytes() {
+    final ByteBuffer result = udf.rpad(BYTES_123, 4, null);
+    assertThat(result, is(nullValue()));
+  }
+
+  @Test
+  public void shouldReturnNullForEmptyPaddingString() {
     final String result = udf.rpad("foo", 4, "");
     assertThat(result, is(nullValue()));
   }
 
   @Test
-  public void shouldPadEmptyInput() {
+  public void shouldReturnNullForEmptyPaddingBytes() {
+    final ByteBuffer result = udf.rpad(BYTES_123, 4, EMPTY_BYTES);
+    assertThat(result, is(nullValue()));
+  }
+
+  @Test
+  public void shouldPadEmptyInputString() {
     final String result = udf.rpad("", 4, "foo");
     assertThat(result, is("foof"));
   }
 
   @Test
-  public void shouldTruncateInputIfTargetLengthTooSmall() {
+  public void shouldPadEmptyInputBytes() {
+    final ByteBuffer result = udf.rpad(EMPTY_BYTES, 4, BYTES_45);
+    assertThat(result, is(ByteBuffer.wrap(new byte[]{4,5,4,5})));
+  }
+
+  @Test
+  public void shouldTruncateInputIfTargetLengthTooSmallString() {
     final String result = udf.rpad("foo", 2, "bar");
     assertThat(result, is("fo"));
   }
 
   @Test
-  public void shouldReturnNullForNegativeLength() {
+  public void shouldTruncateInputIfTargetLengthTooSmallBytes() {
+    final ByteBuffer result = udf.rpad(BYTES_123, 2, BYTES_45);
+    assertThat(result, is(ByteBuffer.wrap(new byte[]{1,2})));
+  }
+
+  @Test
+  public void shouldReturnNullForNegativeLengthString() {
     final String result = udf.rpad("foo", -1, "bar");
     assertThat(result, is(nullValue()));
   }
 
+  @Test
+  public void shouldReturnNullForNegativeLengthBytes() {
+    final ByteBuffer result = udf.rpad(BYTES_123, -1, BYTES_45);
+    assertThat(result, is(nullValue()));
+  }
+
   @Test
   public void shouldReturnEmptyStringForZeroLength() {
     final String result = udf.rpad("foo", 0, "bar");
     assertThat(result, is(""));
   }
 
   @Test
-  public void shouldReturnNullForNullLength() {
+  public void shouldReturnEmptyByteBufferForZeroLength() {
+    final ByteBuffer result = udf.rpad(BYTES_123, 0, BYTES_45);
+    assertThat(result, is(EMPTY_BYTES));
+  }
+
+  @Test
+  public void shouldReturnNullForNullLengthString() {
     final String result = udf.rpad("foo", null, "bar");
     assertThat(result, is(nullValue()));
   }
 
+  @Test
+  public void shouldReturnNullForNullLengthBytes() {
+    final ByteBuffer result = udf.rpad(BYTES_123, null, BYTES_45);
+    assertThat(result, is(nullValue()));
+  }
+
+  private final ByteBuffer BYTES_123 = ByteBuffer.wrap(new byte[]{1,2,3});
```

Comment:
nit: let's keep the member variables together. We can move these to the top of the file where there's already
```
private final RPad udf = new RPad();
```
Also, let's make these static.